nibabel/freesurfer/io.py

```diff
@@ -372,11 +398,101 @@ def read_annot(filepath, orig_ids=False):
+    """Read in an old-style Freesurfer colour table from `fobj`.
+
+    This function is used by :func:`read_annot`.
+
+    Parameters
+    ----------
+
+    fobj : file-like
+        Open file handle to a Freesurfer `.annot` file, with seek point
+        at the beginning of the colour table data.
+    n_entries : int
+        Number of entries in the colour table.
+
+    Returns
+    -------
+
+    ctab : ndarray, shape (n_entries, 5)
+        RGBA colortable array - the last column contains all zeros.
+    names : list of str
+        The names of the labels. The length of the list is n_entries.
+    """
+    dt = _ANNOT_DT
+
+    # orig_tab string length + string
+    length = np.fromfile(fobj, dt, 1)[0]
+    orig_tab = np.fromfile(fobj, '>c', length)
+    orig_tab = orig_tab[:-1]
+    names = list()
+    ctab = np.zeros((n_entries, 5), dt)
+    for i in xrange(n_entries):
+        # structure name length + string
+        name_length = np.fromfile(fobj, dt, 1)[0]
+        name = np.fromfile(fobj, "|S%d" % name_length, 1)[0]
+        names.append(name)
+        # read RGBA for this entry
+        ctab[i, :4] = np.fromfile(fobj, dt, 4)
+
+    return ctab, names
+
+
+def _read_annot_ctab_new_format(fobj, ctab_version):
+    """Read in a new-style Freesurfer colour table from `fobj`.
+
+    This function is used by :func:`read_annot`.
+
+    Parameters
+    ----------
+
+    fobj : file-like
+        Open file handle to a Freesurfer `.annot` file, with seek point
+        at the beginning of the colour table data.
+    ctab_version : int
+        Colour table format version - must be equal to 2
+
+    Returns
+    -------
+
+    ctab : ndarray, shape (n_labels, 5)
+        RGBA colortable array - the last column contains all zeros.
+    names : list of str
+        The names of the labels. The length of the list is n_labels.
+    """
+    dt = _ANNOT_DT
+    # This code works with a file version == 2, nothing else
+    if ctab_version != 2:
+        raise Exception('Unrecognised .annot file version (%i)', ctab_version)
+    # maximum LUT index present in the file
+    max_index = np.fromfile(fobj, dt, 1)[0]
+    ctab = np.zeros((max_index, 5), dt)
+    # orig_tab string length + string
+    length = np.fromfile(fobj, dt, 1)[0]
+    np.fromfile(fobj, "|S%d" % length, 1)[0]  # Orig table path
+    # number of LUT entries present in the file
+    entries_to_read = np.fromfile(fobj, dt, 1)[0]
+    names = list()
+    for _ in xrange(entries_to_read):
+        # index of this entry
+        idx = np.fromfile(fobj, dt, 1)[0]
+        # structure name length + string
+        name_length = np.fromfile(fobj, dt, 1)[0]
+        name = np.fromfile(fobj, "|S%d" % name_length, 1)[0]
+        names.append(name)
+        # RGBA
+        ctab[idx, :4] = np.fromfile(fobj, dt, 4)
+
+    return ctab, names
+
+
+def write_annot(filepath, labels, ctab, names, fill_ctab=True):
     """Write out a Freesurfer annotation file.
 
```

Comment:
Maybe good to note that we exclusively write version 2 ("new-style") .annot files.